### DISCUSSION TITLE
Set cargo size limit per providers

### DISCUSSION THREAD

**Discussion by zellux** [Opinionated-Blueprints/10-Books-Full-of-Rails#100] [Q&A] [answered]
I have some provider stations that generate products slowly, by default the provider train will wait until full cargo before moving to requester stations. Is there a way to set min cargo size in provider station to allow the train to dispatch? E.g. if the item count >= 1000 it will allow requester interrupts to move it to next requester station.

Thanks in advance.

**Comment by Bocian-1** [comment, marked as answer]
What I usually do in those cases is just use a smaller train. I have 2-4 regular trains, but a 1-1 for something like enriched uranium.
The main issue for this right now thanks to making trains as generic as possible, is that if you want to have a more specific condition for a specific train, you have to make it a separate group and modify the conditions for that and there are some other consequences flowing from it like needing a separate refuelling station and depots.
There could be an additional check in the interrupt for a signal produced at the provider if a desired condition is met, but given how specific it is, it doesn't make sense to include it by default.

But on a case-by-case basis you can add this yourself to keep everything in the same group.
In your case:
read train cargo -> compare item count against desired threshold -> send signal to train if threshold is crossed
And in the interrupt add an alternative condition for leaving the provider if the signal is received

**Comment by zellux** [comment]
Thanks for your quick answer, that totally makes sense.
